src/common/overlaycmn.cpp

```diff
@@ -222,9 +218,21 @@ void wxOverlayImpl::EndDrawing(wxDC* WXUNUSED(dc))
 {
 }
 
+#ifndef wxHAS_NATIVE_OVERLAY
 wxOverlay::Impl* wxOverlay::Create()
 {
     return new wxOverlayImpl;
 }
+#endif
+
+#endif // wxHAS_GENERIC_OVERLAY
 
-#endif // !wxHAS_NATIVE_OVERLAY
+wxOverlay::wxOverlay()
+{
+    m_impl = Create();
+#if defined(wxHAS_GENERIC_OVERLAY) && defined(wxHAS_NATIVE_OVERLAY)
+    if (m_impl == NULL)
+        m_impl = new wxOverlayImpl;
+#endif
```

Comment:
This seems a bit confusing to me, could we rather always use just `Create()` here and fallback on creating generic `wxOverlayImpl` in wxGTK code?

This would require renaming this class here to `wxOverlayImplGeneric`, but IMO this would be a good thing too because it's confusing to have 2 classes with the same name in the same program (even if one is inside an anonymous namespace and the other one is not).

Reply:
My preference was to keep the implementations separate and private. But there will likely be further structural changes, so I'm not going to worry too much about this just yet.